tenable_io/client.py

```diff
@@ -46,20 +43,35 @@ def __init__(
         self._endpoint = endpoint
         self._impersonate = impersonate
 
-        self._headers = {
-            u'X-ApiKeys': u'accessKey=%s; secretKey=%s;' % (self._access_key, self._secret_key),
-            u'User-Agent': u'TenableIOSDK Python/%s' % ('.'.join([str(i) for i in sys.version_info][0:3]))
-        }
-
-        if impersonate is not None:
-            self._headers[u'X-Impersonate'] = u'username=%s' % self._impersonate
-
+        self._init_session()
         self._ini_api()
         self._init_helpers()
 
+    def _init_session(self):
+        """
+        Initializes the requests session
+        """
+        retries = Retry(
+                    total=3,
+                    status_forcelist={429}, 
+                    backoff_factor=0.1, 
+                    respect_retry_after_header=True
+        )
+        adapter = requests.adapters.HTTPAdapter(retries)
+        self._session = requests.Session()
+        self._session.mount('https://', adapter)
```

Comment:
We should mount for `http://` as well to not make any assumption on how the client is configured.